src/Sylius/Behat/Context/Ui/Admin/ManagingProductsContext.php

```diff
@@ -0,0 +1,350 @@
+    }
+
+    /**
+     * @Given /^I want to create a new configurable product$/
+     */
+    public function iWantToCreateANewConfigurableProduct()
+    {
+        $this->createConfigurableProductPage->open();
+    }
+
+    /**
+     * @When I specify its code as :code
+     * @When I do not specify its code
+     */
+    public function iSpecifyItsCodeAs($code = null)
+    {
+        $currentPage = $this->currentPageResolver->getCurrentPageWithForm([
+            $this->createSimpleProductPage,
+            $this->createConfigurableProductPage,
+        ]);
+
+        $currentPage->specifyCode($code);
+    }
+
+    /**
+     * @When I name it :name in :language
+     */
+    public function iNameItIn($name, $language)
+    {
+        $currentPage = $this->currentPageResolver->getCurrentPageWithForm([
+            $this->createSimpleProductPage,
+            $this->createConfigurableProductPage,
+        ]);
+
+        $currentPage->nameItIn($name, $language);
+    }
+
+    /**
+     * @When I rename it to :name in :language
+     */
+    public function iRenameItToIn($name, $language)
+    {
+        $currentPage = $this->currentPageResolver->getCurrentPageWithForm([
+            $this->updateSimpleProductPage,
+            $this->updateConfigurableProductPage,
+        ], $this->sharedStorage->get('product'));
+
+        $currentPage->nameItIn($name, $language);
+    }
+
+    /**
+     * @When I add it
+     * @When I try to add it
+     */
+    public function iAddIt()
+    {
+        $currentPage = $this->currentPageResolver->getCurrentPageWithForm([
+            $this->createSimpleProductPage,
+            $this->createConfigurableProductPage,
+        ]);
+
+        $currentPage->create();
+    }
+
+    /**
+     * @When /^I specify its price to ("(?:€|£|\$)[^"]+")$/
+     */
+    public function iMakeItPricedAt($price)
+    {
+        $this->createSimpleProductPage->specifyPrice($price);
+    }
+
+    /**
+     * @Given the product :productName should appear in the shop
+     * @Given the product :productName should be in the shop
+     */
+    public function theProductShouldAppearInTheShop($productName)
+    {
+        $this->iWantToBrowseProducts();
+
+        Assert::true(
+            $this->indexPage->isSingleResourceOnPage(['name' => $productName]),
+            sprintf('The product with name %s has not been found.', $productName)
+        );
+    }
+
+    /**
+     * @When I want to browse products
+     */
+    public function iWantToBrowseProducts()
+    {
+        $this->indexPage->open();
+    }
+
+    /**
+     * @Then I should see :numberOfProducts products in the list
+     */
+    public function iShouldSeeProductsInTheList($numberOfProducts)
+    {
+        $foundRows = $this->indexPage->countItems();
+
+        Assert::eq(
+            $numberOfProducts,
+            $foundRows,
+            sprintf('%s rows with products should appear on page, %s rows has been found', $numberOfProducts, $foundRows)
```

Comment:
You could remove sprintf and `%s rows with products should appear on page, %s rows has been found` will be enough here